w3/streaming-multihash/src/sha256.js

```diff
@@ -0,0 +1,8 @@
+import { sha256 } from '@noble/hashes/sha256'
+import { derive } from './hasher.js'
+
+export const { name, code, create, digestStream } = derive({
+  name: 'sha2-512',
+  code: 0x12,
+  hasher: sha256,
+})
```

Comment:
these wrappers should probably return an fn with the same params as the "create" fn for the hash, call the "create" and send the result (`RawHasher`) to `derive`

https://github.com/paulmillr/noble-hashes#blake2b-blake2s

Reply:
Intention here was that `derive` generates `StreamingMultihashHasher` which is roughly an equivalent of [`MultihashHasher`](https://github.com/multiformats/js-multiformats/blob/bb14a29dd823a517ef0c6c741d265e022591d831/src/hashes/interface.ts#L38-L58)

```js
export interface MultihashHasher<Code extends number = number> {
  /**
   * Takes binary `input` and returns it (multi) hash digest. Return value is
   * either promise of a digest or a digest. This way general use can `await`
   * while performance critical code may asses return value to decide whether
   * await is needed.
   *
   * @param {Uint8Array} input
   */
  digest(input: Uint8Array): Promise<MultihashDigest> | MultihashDigest

  /**
   * Name of the multihash
   */
  name: string

  /**
   * Code of the multihash
   */
  code: Code
}
```